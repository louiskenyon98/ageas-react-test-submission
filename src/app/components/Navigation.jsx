import React from 'react';
import {Link} from 'react-router-dom';
import styles from '../../themes/style.scss';
class Navigation extends React.Component {
    render() {
        return (

            <div className={`ui menu ${styles.nav}`}>
                <Link to="/" className={`header item ${styles.title}`}>notes+</Link>
                <div className="link item right">
                    <Link to="/new"><div className="ui inverted primary button">New Note</div></Link>
                </div>
            </div>
        )
    }
}
export default Navigation;
