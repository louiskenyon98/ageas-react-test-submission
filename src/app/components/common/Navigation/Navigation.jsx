import React from 'react';
import {Link} from 'react-router-dom';

import styles from '../../../../themes/style.scss';
import SelectContainer from '../../../containers/common/formFields/SelectContainer';

class Navigation extends React.PureComponent {
    render() {
        return (
            <div className={`ui menu ${styles.nav}`}>
                <Link to="/" className={`header item ${styles.title}`}>notes+</Link>
                {/*Select*/}
                <SelectContainer />
                <div className="link item right">
                    <Link to="/new">
                        <div className="ui inverted primary button">New Note</div>
                    </Link>
                </div>
            </div>
        )
    }
}

export default Navigation;
