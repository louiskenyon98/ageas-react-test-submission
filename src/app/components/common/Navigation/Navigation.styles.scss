.nav {
  .title {
    font-family: 'Major Mono Display', monospace;
    font-size: 2.5em !important;
    padding: 0.6em !important;
  }
}
