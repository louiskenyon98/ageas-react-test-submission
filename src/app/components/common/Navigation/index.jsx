import React from 'react';
import {Link} from 'react-router-dom';
import styles from '../../../../themes/style.scss';

class Navigation extends React.PureComponent {
    render() {
        return (

            <div className={`ui menu ${styles.nav}`}>
                <Link to="/" className={`header item ${styles.title}`}>notes+</Link>
                <div className="link item right">
                    <div className="ui simple dropdown item">
                        Filter Notes
                        <i className="dropdown icon"></i>
                        <div className="menu">
                            <div className="item">Most recent created first</div>
                            <div className="item">Most recent edit first</div>
                        </div>
                    </div>
                    <Link to="/new">
                        <div className="ui inverted primary button">New Note</div>
                    </Link>
                </div>
            </div>
        )
    }
}

export default Navigation;
