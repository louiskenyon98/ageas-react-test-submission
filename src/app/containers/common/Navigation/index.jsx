import React from 'react';
import {connect} from 'react-redux';
import {onChangeFilterValue} from '../../../actions/filterValues/index.actions';

import Navigation from '../../../components/common/Navigation/Navigation';

class NavigationContainer extends React.Component {
    render() {
        return (
            <Navigation
                filterOptionValue={this.props.filterOption}
                onChangeFilterValue={this.props.onChangeFilterValue}
            />
        )
    }
}

const mapStateToProps = (state) => {
    return {
        filterOption: state.notes.filterOption
    }
};

export default connect(mapStateToProps, {onChangeFilterValue})(NavigationContainer);
