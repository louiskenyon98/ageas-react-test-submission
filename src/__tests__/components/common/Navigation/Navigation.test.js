import React from 'react';
import {shallow} from 'enzyme';
import Navigation from '../../../../app/components/common/Navigation/Navigation';

describe('Navigation', () => {
    let props = {};
    let wrapper = () => shallow(<Navigation {...props}/>);

    describe('render', () => {
        it('should render', () => {
            expect(wrapper()).toMatchSnapshot();
        });
    });
});
