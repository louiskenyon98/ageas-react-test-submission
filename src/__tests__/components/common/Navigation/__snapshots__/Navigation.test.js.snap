// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation render should render 1`] = `
<div
  className="ui menu nav"
>
  <Link
    className="header item title"
    replace={false}
    to="/"
  >
    notes+
  </Link>
  <Connect(SelectContainer) />
  <div
    className="link item right"
  >
    <Link
      replace={false}
      to="/new"
    >
      <div
        className="ui inverted primary button"
      >
        New Note
      </div>
    </Link>
  </div>
</div>
`;
